fe/fe-core/src/main/java/org/apache/doris/statistics/SQLStatisticsTask.java

```diff
@@ -17,47 +17,119 @@
 
 package org.apache.doris.statistics;
 
-import org.apache.doris.analysis.SelectStmt;
+import org.apache.doris.catalog.Database;
+import org.apache.doris.catalog.Env;
+import org.apache.doris.catalog.Table;
+import org.apache.doris.common.DdlException;
+import org.apache.doris.common.InvalidFormatException;
+import org.apache.doris.statistics.StatisticsTaskResult.TaskResult;
+import org.apache.doris.statistics.StatsGranularity.Granularity;
+import org.apache.doris.statistics.util.InternalQuery;
+import org.apache.doris.statistics.util.InternalQueryResult;
+import org.apache.doris.statistics.util.InternalQueryResult.ResultRow;
+import org.apache.doris.statistics.util.InternalSqlTemplate;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
 
 import java.util.List;
+import java.util.Map;
 
 /**
  * A statistics task that collects statistics by executing query.
  * The results of the query will be returned as @StatisticsTaskResult.
  */
 public class SQLStatisticsTask extends StatisticsTask {
-    private SelectStmt query;
+    private String statement;
 
     public SQLStatisticsTask(long jobId, List<StatisticsDesc> statsDescs) {
         super(jobId, statsDescs);
     }
 
     @Override
     public StatisticsTaskResult call() throws Exception {
-        // TODO
-        // step1: construct query by statsDescList
-        constructQuery();
-        // step2: execute query
-        // the result should be sequence by @statsTypeList
-        List<String> queryResultList = executeQuery(query);
-        // step3: construct StatisticsTaskResult by query result
-        constructTaskResult(queryResultList);
-        return null;
+        checkStatisticsDesc();
+        List<TaskResult> taskResults = Lists.newArrayList();
+
+        for (StatisticsDesc statsDesc : statsDescs) {
+            statement = constructQuery(statsDesc);
+            TaskResult taskResult = executeQuery(statsDesc);
+            taskResults.add(taskResult);
+        }
+
+        return new StatisticsTaskResult(taskResults);
     }
 
-    protected void constructQuery() {
-        // TODO
-        // step1: construct FROM by @granularityDesc
-        // step2: construct SELECT LIST by @statsTypeList
+    protected String constructQuery(StatisticsDesc statsDesc) throws DdlException,
+            InvalidFormatException {
+        Map<String, String> params = getQueryParams(statsDesc);
+        List<StatsType> statsTypes = statsDesc.getStatsTypes();
+        StatsType type = statsTypes.get(0);
+
+        StatsGranularity statsGranularity = statsDesc.getStatsGranularity();
+        Granularity granularity = statsGranularity.getGranularity();
+        boolean nonPartitioned = granularity != Granularity.PARTITION;
+
+        switch (type) {
```

Comment:
I think we could finish the collection for these stats in single SQL, such as 

```sql
select * from (select max(col1) as size from t1) as tt, (select count(col1) as num from t1) as ttt;
```

Reply:
It is true that you can collect multiple different statistics in the same statement. However, if you collect too many statistics in a task at the same time, the cluster pressure may be excessive and normal query services may be affected. 

Therefore, statistical tasks are divided into groups according to the type of statistics.

Reply:
you can just limit execution resource by set the memory upperbound and prallel execution instatnce for the stats SQL though, so I think it's not a good reason for that.....

Reply:
In fact, as much as possible, we have put statistics that can be collected together (such as MIN, MAX, NDV) into one group and collected in one SQL. At the same time, it also makes corresponding resource restrictions (such as parallelism, etc.). 

However, if all the aggregated metrics are put into a SQL query and the resources are limited, the collection time may be longer; In addition, when there is a large amount of data, some indicators (e.g., MAX/AVG_COL_LEN) can be collected by sampling, but some indicators (e.g., MIN, MAX, NDV) need to be fully collected, so it may not be appropriate to put them all in one statement.

Also, if the user only wants to quickly collect statistics for one statistical category, then in fact all statistics will be collected. 

Including the problems mentioned above, as well as some other reasons, we have divided the tasks according to some rules~

Reply:
First, I don't think it would take much longer time than separete it to many small SQL, besides put those SQL into single statement actually make the resource control of statistcs more easy, otherwise you should set resource limitations for each statistic SQL and control the total resource occupation of these SQL and should make sure there aren't two many small SQL of statistics executing, all about these is unnessary and make resource management and code maintenance much more complicated.
And, Is really neccessary to enable users setting which concrete metric of column is needed by analyze SQL statement? I think this is totally redundant.

Reply:
I think both ways can work. through testing, we found that when a single SQL collects multiple metrics, its query time  increases proportionally with the number of metrics. For NDV, if the number of rows is 100 million, it takes 3s. When there is a large amount of data and a lot of columns, it takes longer. The current strategy is collecting statistics in parallel. Test data from @EmmyMiao87 
docs：https://docs.google.com/document/d/1u1L6XhyzKShoyYRwFQ6kE1rnvY2iFwauwg289au5Qq0/edit